Sources_v3/Workers/Background/MessageEditor.swift

```diff
@@ -57,58 +54,43 @@ class MessageEditor<ExtraData: ExtraDataTypes>: Worker {
         let wasEmpty = pendingMessageIDs.isEmpty
         changes.pendingEditMessageIDs.forEach { pendingMessageIDs.insert($0) }
         if wasEmpty {
-            database.backgroundReadOnlyContext.perform { [weak self] in
-                self?.processNextMessage()
-            }
+            processNextMessage()
         }
     }
 
     private func processNextMessage() {
         guard let messageId = pendingMessageIDs.first else { return }
         
-        guard let dto = database.backgroundReadOnlyContext.message(id: messageId) else {
-            removeMessageIDAndContinue(messageId)
-            return
-        }
-        
-        guard dto.localMessageState == .pendingSync else {
-            removeMessageIDAndContinue(messageId)
-            return
-        }
-        
-        markMessage(withID: messageId, as: .syncing) {
+        backgroundContext.perform { [weak self] in
+            guard let dto = self?.backgroundContext.message(id: messageId), dto.localMessageState == .pendingSync else {
+                self?.removeMessageIDAndContinue(messageId)
+                return
+            }
+            
             let requestBody = dto.asRequestBody() as MessageRequestBody<ExtraData>
-            self.apiClient.request(endpoint: .editMessage(payload: requestBody)) {
-                switch $0 {
-                case .success:
-                    self.markMessage(withID: messageId, as: nil) {
-                        self.removeMessageIDAndContinue(messageId)
-                    }
-                case .failure:
-                    self.markMessage(withID: messageId, as: .syncingFailed) {
-                        self.removeMessageIDAndContinue(messageId)
-                    }
+            
+            self?.markMessage(withID: messageId, as: .syncing)
```

Comment:
There are plenty of `self?` statements but that's what we expect, I think. 

If we say `guard let self = self else { return }` in the request completion, the worker's lifetime can exceed the current user's session and some unexpected database write operations can happen before it finally get's deallocated

Reply:
In the same light, if we don't hold a strong ref to self here (and always use `self?`), worker can be deallocated during the middle of processing, so after marking a message as syncing (line 72) but before being able to mark it as synced?

Reply:
If we call `setUser` with the same user-id as the current one, we don't need to flush the database nor recreate background workers -> workers will not be deallocated and messages will be updated.

If we change the user, the database will be flushed -> there will be no messages from the previous user -> we don't care about message local states